File: js_solutions/1287_element_appearing_more_than_25_in_sorted_array.js
/**
 * @link https://leetcode.com/problems/element-appearing-more-than-25-in-sorted-array/
 * @description Given an integer array sorted in non-decreasing order
 * There is exactly one integer in the array that occurs more than 25% of the time,
 * return that integer.
 * @param {Array []} nums
 * @returns {Number}
 */
const findSpecialInteger = (nums) => {
  // Find 25 %
  const quarter = Math.trunc(nums.length / 4);
  const frequency = {};
  // Find required element in the array
  for (const num of nums) {
    frequency[num] = (frequency[num] || 0) + 1;
    if (frequency[num] > quarter) {
      return num;
    }
  }
};

File: python_solutions/1287_element_appearing_more_than_25_in_sorted_array.py
def findSpecialInteger(self, arr: List[int]) -> int:
  # Find 25 % of the given array's length
  quarter = len(arr) // 4
  # Find required element in the given array
  for i in range(len(arr)):
    if(arr[i] == arr[i + quarter]):
      return arr[i]
        